tests/basic_tests.cpp

```diff
@@ -17,6 +17,7 @@ TYPED_TEST(basic_tests, insane_url) {
   ASSERT_EQ(r->get_hostname(), "");
   ASSERT_EQ(r->get_port(), "");
   ASSERT_EQ(r->get_pathname(), "@EEEEEEEEEE");
+  ASSERT_FALSE(url->set_port("invalid80"));
```

Comment:
This is perfect but your test will fail on Ada::url. You changed url_aggregator set_port implementation but not the other one.